modules/dnn/include/opencv2/dnn/dnn.hpp

```diff
@@ -1296,6 +1296,20 @@ CV__DNN_INLINE_NS_BEGIN
           */
          CV_WRAP DetectionModel(const Net& network);
 
+         /**
+          * @brief nmsAcrossClasses defaults to false,
+          * such that when non max suppression is used during the detect() function, it will do so per-class.
+          * This function allows you to toggle this behaviour.
+          * @param[in] value The new value for nmsAcrossClasses
+          */
+         CV_WRAP void setNmsAcrossClasses(bool value);
```

Comment:
Please change return type:
`void` => `DetectionModel&` (`return *this;`)
to allow chains of `.set*()` calls.

(+ default dummy ctor `DetectionModel()` for bindings (see mentioned PR), + checks for nullptr impl from this ctor)

Reply:
Right, forgot about this. Would you be able to elaborate further you mean by "checks for nullptr impl", please? In other words,

```
DetectionModel::DetectionModel() : Model()
{
    // What goes here?
}
```

I understand the implementation field `impl` will be null in this constructor, but I am not sure what exactly to put here. I just updated this PR with a new commit to show the differences I made. I will change and squash it later once we rectify this.

Reply:
Constructor is fine.
Need to add checks (`CV_Assert()`) into other methods (where `impl` is nullptr and `impl.dynamicCast<DetectionModel_Impl>()` is nullptr too).

Reply:
Added another commit with the assertions, is this what you meant? Also added the comments next to them, let me know if I should remove or reword